sympy/solvers/solveset.py

```diff
@@ -304,6 +305,14 @@ def inv(trig):
                 invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
             return _invert_real(f.args[0], invs, symbol)
 
+    if isinstance(f, LambertW):
+        return _invert_real(f.args[0], imageset(Lambda(n, n*exp(n)), g_ys), symbol)
+
+    if _is_lambert(f - g_ys.args[0], symbol):
+        soln = _solve_as_lambert(f, g_ys.args[0], symbol, domain=S.Reals)
+        if soln is not None:
+            return symbol, soln
```

Comment:
Most of the other codepaths in this function end with `return _invert_real`. Why doesn't that happen in the `_is_lambert` case?

Reply:
> Most of the other codepaths in this function end with return _invert_real. Why doesn't that happen in the _is_lambert case?

Since it's very different from others.
It has many ways of representation. ( discussed earlier in detail)

Reply:
And this is the same reason I was reluctant to add code into the `invert` function.

Reply:
Can it be made more like invert though?

Really this shouldn't be any different from inverting other functions...

Reply:
I think it could have probably a very different approach and should be carried out in a separate PR. As of now I think we should put a comment over the lines saying ' it should be made into more like other `invert` inverse '